AHK/SteamBulkKeyActivator.ahk

```diff
@@ -307,27 +407,41 @@ applog("OS Version  =>" . A_OSVersion)
 applog("is x64      =>" . A_Is64bitOS)
 applog("is elevated =>" . A_IsAdmin)
 applog(" ---- console log -----")
-MsgBox, 64, Automation, Please do not touch the keyboard or mouse while the macro is running.`n(press escape to stop the program at any time)
+;MsgBox, 64, Automation, Please do not touch the keyboard or mouse while the macro is running.`n(press escape to stop the program at any time)
+IfExist, %A_WorkingDir%\activation.log
+{
+	FileDelete, %A_WorkingDir%\activation.log
+
+}
 IfExist, %A_WorkingDir%\keys.txt
 {
 	applog("found keys.txt")
-	Loop, read, %A_WorkingDir%\keys.txt
-	{
-		Loop, parse, A_LoopReadLine, %A_Tab%
-		{
-			steam_activate_key(A_LoopField)
-			Sleep,1000
-		}
-	}
-	MsgBox, 64, Success!, We looped through the entire keys.txt file.
-	run notepad %A_WorkingDir%\activation.log
-
+	FileRead, keytext, %A_WorkingDir%\keys.txt
+	MsgBox, 64, Automation, Please do not touch the keyboard or mouse while the macro is running.`n(press escape to stop the program at any time)
 }else{
-	;add a keys.txt file plz
-	applog("we did not find any keys in keys.txt ! ")
-	MsgBox, 48, No keys.txt found !, Please add a keys.txt file in the root of this program.`n1 key per line.`nxxxx-xxxx-xxxx-xxxx
-	run notepad %A_WorkingDir%\keys.txt
+	;read keys from edit box
+	keytext:=MultiLineInput("Enter keys here or into keys.txt and restart.")
+
 }
+linecount=0
+keycount=0
+Loop, parse, keytext, `n, `r
+{
+	;will only activate keys in the format XXXXX-XXXXX-XXXXX
+	;this is mostly for easy pasting from humble bundle and indiegala key pages
+	if(RegExMatch(A_LoopField,"^\w{5}-\w{5}-\w{5}$")){ 
```

Comment:
please make it also look inbetween other text and catch multiple keys per line.
also there can't be more then 1 instance of the same char in a row!

Reply:
I stopped using this script and switched to https://github.com/Ezzpify/SteamBulkActivator.
It uses the Steam API to activate keys, so it's faster and it won't be broken by UI changes.